src/System.Data.SQLite/NativeMethods.cs

```diff
@@ -127,6 +127,56 @@ internal static class NativeMethods
 		[DllImport(c_dllName, CallingConvention = c_callingConvention)]
 		public static extern int sqlite3_trace_v2(SqliteDatabaseHandle db, SQLiteTraceEvents eventsMask, SQLiteTraceV2Callback callback, IntPtr userData);
 
+#if NET5_0
+		static NativeMethods()
+		{
+			var assembly = typeof(NativeMethods).Assembly;
+			var configFile = System.IO.Path.GetFileName(assembly.Location) + ".config";
+			if (System.IO.File.Exists(configFile))
+			{
+				try
+				{
+					using var stream = new System.IO.FileStream(configFile, System.IO.FileMode.Open);
+					using var reader = System.Xml.XmlReader.Create(stream);
+					var dllMapConfig = System.Xml.Linq.XDocument.Load(reader);
+
+					
+					foreach (var element in dllMapConfig.Root.Elements())
+					{
+						if (element.Name != "dllmap")
+							continue;
+
+						if (element.Attribute("dll").Value == c_dllName)
+						{
+							s_mappedDllName = element.Attribute("target").Value;
+							break;
+						}
+					}
+
+					if (!string.IsNullOrEmpty(s_mappedDllName))
+						NativeLibrary.SetDllImportResolver(typeof(NativeMethods).Assembly, DllImportResolver);
+				}
+				catch (Exception)
+				{
+				}
+			}
+		}
+
+		private static IntPtr DllImportResolver(string libraryName, System.Reflection.Assembly assembly, DllImportSearchPath? searchPath)
+		{
+			var handle = IntPtr.Zero;
+
+			if (libraryName == c_dllName && !string.IsNullOrEmpty(s_mappedDllName))
+				NativeLibrary.TryLoad(s_mappedDllName, out handle);
```

Comment:
Should we throw if this fails?

Reply:
The argument could be made for both cases. The default behavior will be to not use a custom library name and try to load the native library based on `sqlite3`.

Reply:
I'd think if the client has specified a `config` file that maps `sqlite3`, they'd like to know immediately if the library is missing, rather than with mysterious runtime errors when the wrong native library is used.

Reply:
Changed.